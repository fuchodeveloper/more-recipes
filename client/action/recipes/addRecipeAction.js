import { batchActions } from 'redux-batched-actions';
import axios from 'axios';
import alertify from 'alertify.js';
import { ADD_RECIPE, ADD_RECIPE_ERROR } from '../types';
import { setFetching, unsetFetching } from '../fetching';

/**
 * @description create a recipe
 *
 * @export addRecipeActionCreator
 *
 * @param {Object} recipe
 *
 * @returns {Object} recipe
 */
const addRecipeActionCreator = recipe => ({
  type: ADD_RECIPE,
  recipe
});

/**
 * @description create a recipe error
 *
 * @export addRecipeActionError
 *
 * @param {Object} error
 *
 * @returns {Object} error
 */
const addRecipeActionError = error => ({
  type: ADD_RECIPE_ERROR,
  error
});

const addRecipeAction = recipe => (dispatch) => {
  dispatch(setFetching());
  return axios.post('/api/v1/recipes', recipe)
    .then((response) => {
      alertify.delay(900);
      alertify.logPosition('bottom right');
      alertify.success(response.data.message);
      dispatch(batchActions([
        dispatch(addRecipeActionCreator(response.data.recipe)),
        unsetFetching()
      ]));
    })
    .catch((error) => {
      dispatch([
        dispatch(addRecipeActionError(error.response.data.error)),
        unsetFetching()
      ]);
    });
};

export default addRecipeAction;
