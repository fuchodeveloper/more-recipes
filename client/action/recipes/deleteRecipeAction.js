import axios from 'axios';
import alertify from 'alertify.js';
import { DELETE_RECEIPE, DELETE_RECEIPE_FAIL } from '../types';
import networkError from '../networkError';

export const deleteRecipeActionError = error => ({
  type: DELETE_RECEIPE_FAIL,
  error
});

export const deleteRecipeActionCreator = recipes => ({
  type: DELETE_RECEIPE,
  recipes
});

const deleteRecipeAction = recipeId => dispatch =>
  axios.delete(`/api/v1/recipes/${recipeId}`)
    .then((response) => {
      alertify.delay(900);
      alertify.logPosition('bottom right');
      alertify.success(response.data.message);
      dispatch(deleteRecipeActionCreator(response.data.recipes));
    })
    .catch((error) => {
      if (!error.response) {
        return networkError(error);
      }
      dispatch(deleteRecipeActionError(error.response.data));
    });

export default deleteRecipeAction;
