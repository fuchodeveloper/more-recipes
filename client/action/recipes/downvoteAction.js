import axios from 'axios';
import alertify from 'alertify.js';
import { GET_DOWNVOTE } from '../types';

/**
 * Recipe doenvote action creator
 * @param {recipe} recipe
 * @returns {recipe} recipe
 */
export const downvoteRecipeActionCreator = recipe => ({
  type: GET_DOWNVOTE,
  recipe
});

/**
 * Authenticated user can downvote a recipe
 *
 * @export func
 * @param {integer} param
 * @returns {object} object
 */
const downvoteRecipe = param =>
  dispatch => axios.post(`/api/v1/recipes/${param}/downvote`)
    .then((response) => {
      alertify.delay(900);
      alertify.logPosition('bottom right');
      alertify.success(response.data.message);
      return dispatch(downvoteRecipeActionCreator(response.data.recipe));
    })
    .catch((error) => {
      alertify.logPosition('bottom right');
      alertify.error(error.message);
    });


export default downvoteRecipe;
