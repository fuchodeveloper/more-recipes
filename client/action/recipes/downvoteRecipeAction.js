import axios from 'axios';
import alertify from 'alertify.js';
import { DOWNVOTE_RECIPE, DOWNVOTE_RECIPE_ERROR } from '../types';
import networkError from '../networkError';

/**
 * @description Recipe downvote action creator
 *
 * @param {recipe} recipe recipe parameter
 *
 * @returns {recipe} recipe downvoted
 */
export const downvoteRecipeActionCreator = recipe => ({
  type: DOWNVOTE_RECIPE,
  recipe
});
/**
 * @description Recipe downvote error
 *
 * @param {Object} error error parameter
 *
 * @returns {Object} error returns the error object
 *
 */
const downvoteRecipeActionError = error => ({
  type: DOWNVOTE_RECIPE_ERROR,
  error
});

/**
 * @description Authenticated user can downvote a recipe
 *
 * @export downvoteRecipe
 *
 * @param {Number} id
 *
 * @returns {object} downvoted recipe object
 */
const downvoteRecipeAction = id =>
  dispatch => axios.post(`/api/v1/recipes/${id}/downvote`)
    .then((response) => {
      alertify.delay(900);
      alertify.logPosition('bottom right');
      alertify.success(response.data.recipe.message);
      return dispatch(downvoteRecipeActionCreator(response.data.recipe));
    })
    .catch((error) => {
      if (!error.response) {
        return networkError(error);
      }
      alertify.delay(1000);
      alertify.logPosition('bottom right');
      alertify.error('Please login to downvote recipe');
      dispatch(downvoteRecipeActionError(error.response.data));
    });


export default downvoteRecipeAction;
