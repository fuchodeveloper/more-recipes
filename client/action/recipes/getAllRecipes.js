import axios from 'axios';
import { batchActions } from 'redux-batched-actions';
import { GET_ALL_RECIPES, GET_PAGE_DETAILS } from '../types';
import { setFetching, unsetFetching } from '../fetching';

export const receiveRecipes = recipes => ({
  type: GET_ALL_RECIPES,
  recipes
});

export const recipePageDetails = pageCount => ({
  type: GET_PAGE_DETAILS,
  pageCount
});

const getAllRecipes = page => (dispatch) => {
  dispatch(setFetching());
  axios.get(`/api/v1/recipes?page=${page}`)
    .then((response) => {
      dispatch(batchActions([
        receiveRecipes(response.data.recipes),
        recipePageDetails(response.data.pageCount),
        unsetFetching(),
      ]));
    })
    .catch((error) => {
      // dispatch(batchActions([

      // ]));
      console.log(error.message);
    });
};

export default getAllRecipes;
