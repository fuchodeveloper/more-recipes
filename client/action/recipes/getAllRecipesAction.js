import axios from 'axios';
import { batchActions } from 'redux-batched-actions';
import
{ GET_ALL_RECIPES, GET_PAGE_DETAILS, GET_ALL_RECIPES_ERROR } from '../types';
import { setFetching, unsetFetching } from '../fetching';

/**
 * @description get all recipes action creator
 *
 * @param {Object} recipes recipes object parameter
 *
 * @returns {Object} recipes returns all recipes object
 */
const receiveRecipes = recipes => ({
  type: GET_ALL_RECIPES,
  recipes
});

/**
 * @description get all recipes error action
 *
 * @param {Object} error recipes error object
 *
 * @returns {Object} error returns get all recipes error object
 */
const receiveRecipesError = error => ({
  type: GET_ALL_RECIPES_ERROR,
  error
});

/**
 * @description handles page detail
 *
 * @param {Number} pageCount recipe page count parameter
 *
 * @returns {Number} returns recipes page Count
 */
const recipePageCount = pageCount => ({
  type: GET_PAGE_DETAILS,
  pageCount
});

const getAllRecipesAction = page => (dispatch) => {
  dispatch(setFetching());
  return axios.get(`/api/v1/recipes?page=${page}`)
    .then((response) => {
      dispatch(batchActions([
        dispatch(receiveRecipes(response.data.recipes)),
        recipePageCount(response.data.pageCount),
        unsetFetching(),
      ]));
    })
    .catch((serverError) => {
      dispatch(receiveRecipesError(serverError.recipes.data.error));
    });
};

export default getAllRecipesAction;
