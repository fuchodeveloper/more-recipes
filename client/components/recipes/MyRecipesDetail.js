import React from 'react';
import { Link } from 'react-router-dom';
import noodles from '../../assets/img/noodles.jpg';

class MyRecipesDetail extends React.Component {
  render() {
    const { details } = this.props;

    return (
      <div className="col-md-4 mb-2">

        <div className="card mt-1">
            {/* <img className="card-img-top" src="./img/noodles.jpg" alt="Card image cap" /> */}
            <img className="card-img-top" max-width="348px" height="231px" src={details.recipeImage === '' ? noodles : details.recipeImage} alt={details.recipeName}/>
            <div className="card-body">
              <h4 className="card-title">{details.recipeName}</h4>
              <p className="card-text">{`${details.recipeDirection.slice(0, 100)}...`}</p>
              <div className="card-footer custom-card-footer-bg">
                  <p className="card-text"><small className="text-muted">{details.views} <i className="fa fa-eye" aria-hidden="true"/> . {details.upVotes} <i className="fa fa-thumbs-up" aria-hidden="true"/> . {details.favoriteCount} <i className="fa fa-star" aria-hidden="true"/></small></p>
                  <Link to={`/recipes/${details.id}`} className="btn btn-primary btn-primary-color">View Recipe</Link> &nbsp; <Link to={`/edit_recipe/${details.id}`} className="btn btn-light">Edit Recipe</Link>
              </div>
            </div>
        </div>

      </div>
    );
  }
}

export default MyRecipesDetail;
