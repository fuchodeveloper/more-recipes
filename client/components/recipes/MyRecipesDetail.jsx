import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export const MyRecipesDetail = props => (
  <div className="col-md-4 mb-2" id="my-recipe-card">
    <div className="card mt-1">
      <img
        className="card-img-top"
        max-width="348px"
        height="231px"
        src={props.details.image === '' ?
        'https://res.cloudinary.com/fuchodeveloper/image/upload/'
              + 'v1516760699/noodles_c6ltkq.jpg' : props.details.image}
        alt={props.details.name}
      />
      <div className="card-body">
        <h4 className="card-title">
          {`${props.details.name.slice(0, 40)}...`}
        </h4>
        <p className="card-text">
          {`${props.details.direction.slice(0, 100)}...`}
        </p>
        <div className="card-footer custom-card-footer-bg">
          <p className="card-text">
            <small className="text-muted">
              {props.details.views}
            &nbsp;
              <i className="fa fa-eye" aria-hidden="true" /> .
              &nbsp; {props.details.upVotes} &nbsp;
              <i className="fa fa-thumbs-up" aria-hidden="true" /> .
              &nbsp; {props.details.favoriteCount} &nbsp;
              <i className="fa fa-star" aria-hidden="true" />
            </small>
          </p>
          <Link
            to={`/recipes/${props.details.id}`}
            className="btn btn-primary btn-primary-color"
          >
          View
          </Link> &nbsp;
          <a
            href="#"
            onClick={props.onConfirm}
            className="btn btn-danger"
            data-id={props.details.id}
          >Delete
          </a>&nbsp;
          <Link
            to={`/update/${props.details.id}`}
            className="btn btn-light"
            id="recipe-edit"
          >Edit
          </Link>
        </div>
      </div>
    </div>
  </div>
);

MyRecipesDetail.defaultProps = {
  onConfirm: () => {},
  details: {
    views: 0,
    id: 1,
    upVotes: 0,
    favoriteCount: 0
  }
};

MyRecipesDetail.propTypes = {
  onConfirm: PropTypes.func,
  details: PropTypes.shape({
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    direction: PropTypes.string.isRequired,
    views: PropTypes.number,
    id: PropTypes.number,
    upVotes: PropTypes.number,
    favoriteCount: PropTypes.number,
  }),
};

export default MyRecipesDetail;
