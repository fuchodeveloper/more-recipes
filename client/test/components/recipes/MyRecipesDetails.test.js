import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { MyRecipesDetail } from '../../../components/recipes/MyRecipesDetail';

let props;

const setup = () => {
  props = {
    details: {
      image: 'https://res.cloudinary.com/fuchodeveloper/image/upload/'
      + 'v1516760699/noodles_c6ltkq.jpg',
      name: 'jollof rice',
      direction: 'cook for five minutes'
    }
  };
  return shallow(<MyRecipesDetail {...props} />);
};

describe('MyRecipesDetail component', () => {
  describe('MyRecipesDetail component snapshot', () => {
    it('should render with right amount of elements', () => {
      const { wrapper } = setup();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
