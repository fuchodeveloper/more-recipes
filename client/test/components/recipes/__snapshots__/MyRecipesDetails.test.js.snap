// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRecipesDetail component MyRecipesDetail component snapshot should render with right amount of elements 1`] = `undefined`;
